pkg/filter/apiaggregator/apiaggregator.go

```diff
@@ -173,39 +189,41 @@ func (aa *APIAggregator) handle(ctx context.HTTPContext) (result string) {
 	}
 
 	wg := &sync.WaitGroup{}
-	wg.Add(len(aa.spec.APIProxies))
+	wg.Add(len(aa.spec.Pipelines))
 
-	httpResps := make([]context.HTTPResponse, len(aa.spec.APIProxies))
-	for i, proxy := range aa.spec.APIProxies {
-		req, err := aa.newHTTPReq(ctx, proxy, buff)
+	httpRsps := make([]context.HTTPResponse, len(aa.spec.Pipelines))
+	for i, p := range aa.spec.Pipelines {
+		req, err := aa.newHTTPReq(ctx, p, buff)
 		if err != nil {
-			logger.Errorf("BUG: new HTTPProxy request failed %v proxyname[%d]", err, aa.spec.APIProxies[i].HTTPProxyName)
+			logger.Errorf("BUG: new HTTP request failed: %v, pipelinename: %s", err, aa.spec.Pipelines[i].Name)
 			ctx.Response().SetStatusCode(http.StatusBadRequest)
 			return resultFailed
 		}
 
 		go func(i int, name string, req *http.Request) {
 			defer wg.Done()
-			copyCtx, err := aa.newCtx(ctx, req, buff)
-			if err != nil {
-				httpResps[i] = nil
-				return
-			}
-
-			handler, exists := aa.muxMapper.GetHandler(name)
-
+			handler, exists := aa.rctc.GetHTTPPipeline(name)
 			if !exists {
-				httpResps[i] = nil
+				logger.Errorf("pipeline: %s not found in default ns", name)
+				httpRsps[i] = nil
 			} else {
+				w := httptest.NewRecorder()
+				copyCtx := context.New(w, req, tracing.NoopTracing, "no trace")
 				handler.Handle(copyCtx)
-				httpResps[i] = copyCtx.Response()
+				rsp := copyCtx.Response()
+
+				if rsp != nil && rsp.StatusCode() == http.StatusOK {
+					httpRsps[i] = rsp
+				} else {
+					httpRsps[i] = nil
+				}
 			}
```

Comment:
propose change these lines to below:
```go
			handler, exists := aa.rctc.GetHTTPPipeline(name)
			if !exists {
				logger.Errorf("pipeline: %s not found in current namespace", name)
				return
			}

			w := httptest.NewRecorder()
			copyCtx := context.New(w, req, tracing.NoopTracing, "no trace")
			handler.Handle(copyCtx)
			rsp := copyCtx.Response()

			if rsp != nil && rsp.StatusCode() == http.StatusOK {
				httpRsps[i] = rsp
			}
```

Reply:
The difference here is whether we treat `not pipeline exist` as a fatal error. 
Actually, the aggregator has a `partialSucceed` switch, which means the aggregator can accept some API failure situations if it is turned on. 
So I prefer to keep `no pipeline exists` as a normal error in Aggregator since we had a `partialSucceed` switch.

Reply:
sorry, I made a mistake.

but still propose change the log:
```go
logger.Errorf("pipeline: %s not found in current namespace", name)
```

and remove
```go
httpRsps[i] = nil
```
as `nil` is the default value.

Reply:
the early return happens in the closure function, does not impact `partialSucceed`

Reply:
Oh, it's my mistake. Your suggestion is better than mine. Let me update it.

Reply:
Updated